lisa/tools/lsblk.py

```diff
@@ -274,21 +289,37 @@ def get_disks(self, force_run: bool = False) -> List[DiskInfo]:
 
         return disks
 
-    def find_disk_by_mountpoint(
-        self, mountpoint: str, force_run: bool = False
-    ) -> DiskInfo:
+    def _find_by_mountpoint(
+        self, mountpoint: str, force_run: bool = False, find_partition: bool = False
+    ) -> Optional[Union[DiskInfo, PartitionInfo]]:
         disks = self.get_disks(force_run=force_run)
         for disk in disks:
-            # check if disk is mounted and mountpoint matches
             if disk.mountpoint == mountpoint:
-                return disk
+                return disk if not find_partition else None
 
-            # check if any of the partitions is mounted and mountpoint matches
             for partition in disk.partitions:
                 if partition.mountpoint == mountpoint:
-                    return disk
+                    return partition if find_partition else disk
+
+        return None
 
-        raise LisaException(f"Could not find disk with mountpoint {mountpoint}")
+    def find_disk_by_mountpoint(
+        self, mountpoint: str, force_run: bool = False
+    ) -> DiskInfo:
+        result = self._find_by_mountpoint(mountpoint, force_run, find_partition=False)
+        if result is None or isinstance(result, PartitionInfo):
```

Comment:
`or isinstance(result, PartitionInfo)` is not needed. You can rely on above logic to make sure it returns the right type.

Reply:
This will not applicable after taking your suggested approach for `find_partition_by_mountpoint`